### PR TITLE
[json-spirit] Add initial port

### PR DESCRIPTION
**Pull request by smlee-hdactech**
I added the porting job for json-spirit.
Have a nice day.

### CODE REVIEW

File: ports/json-spirit/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#   VCPKG_LIBRARY_LINKAGE     = target library linkage type (static, dynamic)
+#   VCPKG_ROOT_DIR            = <C:\path\to\current\vcpkg>
+#   VCPKG_TARGET_ARCHITECTURE = target architecture (x64, x86, arm)
+#
+
+include(vcpkg_common_functions)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO smlee-hdactech/json_spirit
```
Comment (Codiferous) [inline ports/json-spirit/portfile.cmake]: This appears to be a fork of the library. We try to use the "official" upstream for every package in vcpkg unless there's an excellent reason not to. 

It seems that the most appropriate upstream to use for this is https://github.com/png85/json_spirit, based on what system package managers like Homebrew and Debian use.

Could you update this PR to use a more appropriate upstream?

File: ports/json-spirit/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    REF 28d747f0be5c2f2f8cbfb9a01a2d6d560eec6c80
+    SHA512 ec52a8ca78d294d85a0f3c81272f62e15d0cfc363f9fba4730705ab0b0c14ce38eb11da0ebc9ccba07ecda0ba8944d20aa55bcd907c803e3995d462bb06e0745
+    HEAD_REF master
+)
+
+#set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/json_spirit-4.1.0)
```
Comment (Codiferous) [inline ports/json-spirit/portfile.cmake]: I believe these comments were not intentionally left here. Could you remove them when you update the upstream?

File: ports/json-spirit/portfile.cmake (hunk 1 above)
Comment (smlee-hdactech) [reply ports/json-spirit/portfile.cmake]: O.K. I got it. I will check it using original stream.

File: ports/json-spirit/portfile.cmake (hunk 2 above)
Comment (smlee-hdactech) [reply ports/json-spirit/portfile.cmake]: Of course, I will remove it.

File: ports/json-spirit/portfile.cmake (hunk 1 above)
Comment (smlee-hdactech) [reply ports/json-spirit/portfile.cmake]: I did pull-request to original stream, png85, however, the repository maybe updates no longer.
How about my creating new repository for json-spirit?

Comment (Codiferous) [reply ports/json-spirit/portfile.cmake]: We try to use the upstream repo as much as possible with vcpkg. I've made the changes to this PR to use the png85 repo and modified the PR to only allow static linking. When/if the upstream repo accepts your PR, we can modify our port to reflect the DLL use case.

**Comment by Codiferous** [comment]
I've modified this PR to pull from the png85 repo because that is the source that other repositories use. This means that until your PR containing changes to enable dllexport on some of the entities is accepted in png85, vcpkg will only allow static linkage for json-spirit. 

If your PR is not accepted due to the mainstream repo being inactive, please submit another PR to vcpkg that contains patch files to apply to the json-spirit source. 

Thanks!

**Pull request merged**